rocketc/rocketc.py

```diff
@@ -718,10 +714,15 @@ def _get_list_groups(self, user_id, auth_token, **kwargs):
         """
         api = self.api_rocket_chat
         groups = api.list_all_groups(user_id, auth_token, **kwargs)
+        team, topic = generate_team_variables(self._get_team(self.user_data["username"]))
         if groups["success"]:
-            groups = groups["groups"]
-            for group in groups:
-                yield group["name"]
+            for group in groups["groups"]:
+                fields = group.get("customFields", {})
+                try:
+                    if (team == fields["team"] and topic == fields["topic"] and self.course_id == fields["course"]):
+                            yield group["name"]
+                except IndexError:
```

Comment:
If you want to catch a missing key from a dict, the correct error is a KeyError not an IndexError. @andrey-canon

Reply:
changed